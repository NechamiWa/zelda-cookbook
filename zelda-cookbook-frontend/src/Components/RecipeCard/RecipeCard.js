// eslint-disable-next-line
import React from 'react';
import './RecipeCard.css';
import PropTypes from 'prop-types';
import heart from '../../assets/heart.png';
import greenRupee from '../../assets/GreenRupee.png';

const RecipeCard = (props) => {
  const {
    id,
    image,
    name, 
    hearts, 
    type_image,
    duration, 
    notes,    
    strength, 
    resale
  } = props;

  const clicked = (e) => {
    const recipeTarget = e.target;
    const targetArticle = recipeTarget.closest('article');
    if(targetArticle.className.includes('active-recipe')) {
      targetArticle.classList.remove('active-recipe');
    } else {
      targetArticle.classList.add('active-recipe');  
    }
  };

  const renderIngredients = () => {
    let ingredients = [];
    for (let i = 1; i <= 5; i++ ) {
      if (props[`ingredient${i}_image`]) {
        ingredients.push (
          <h3 key={i} className={`ingredient ${name} ${props[`ingredient${i}`]}`}>
            <img src={ props[`ingredient${i}_image`] } alt={ props[`ingredient${i}`] } />
            { props[`ingredient${i}`] }
          </h3>
        );
      }
    }
    return ingredients;
  };

  const renderTypeImage = () => {
    if (type_image) {
      return <img src={type_image} alt='Recipe Type Icon' className='type_image'/>;
    }
  };

  const renderNotes = () => {
    if (notes) {
      return <h3 className={`${ name } recipe-notes`}>Notes: { notes }</h3>;
    }
  };
  
  return (
    <div className="recipe-card-container">
      <article id={ id } className={ name } onClick={ clicked } >

        <div className={`${ name } recipe-details`}>
          <img src={ image } alt={ name } />
          <div className={`${ name } name-hearts`}>
            <h2 className={`${ name } recipe-name`}>{ name }</h2>
            <h3 className={`${ name } recipe-hearts`}>
              <img src={ heart } alt="Number Of Hearts"/>
              <span id="targetHeart">{ hearts }</span>
              <img id="green-rupee" src={ greenRupee } alt="Green Rupees Jewel"/>
              <p>{ resale }</p>
            </h3>
          </div>
          { renderTypeImage() }
        </div>

        { renderNotes() }

        <div className={`${ name } ingredients`}>
          { renderIngredients() }  
        </div>

        <div className={`${ name } recipe-specific-details`}>
          <h5 className={`${ name } strength`}>{ strength === null ? '' : 'Strength: ' + strength }</h5>
          <h5 className={`${ name } duration`}>{ duration === null ? '' : 'Duration: ' + duration }</h5> 
        </div>
      </article>
    </div>
  );
};

const { number, string, func } = PropTypes;

RecipeCard.propTypes = {
  id: number,
  name: string, 
  hearts: string, 
  type: string,
  type_image: string,
  duration: string, 
  notes: string,
  image: string,
  ingredient1: string, 
  ingredient2: string, 
  ingredient3: string, 
  ingredient4: string, 
  ingredient5: string,
  ingredient1_image: string, 
  ingredient2_image: string, 
  ingredient3_image: string, 
  ingredient4_image: string, 
  ingredient5_image: string,
  ingredient1_id: number, 
  ingredient2_id: number, 
  ingredient3_id: number, 
  ingredient4_id: number, 
  ingredient5_id: number, 
  strength: string, 
  resale: string,
  clicked: func
};

export default RecipeCard;
