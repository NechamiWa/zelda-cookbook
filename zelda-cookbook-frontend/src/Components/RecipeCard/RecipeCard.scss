@import url('https://fonts.googleapis.com/css?family=PT+Serif:700');

$recipe-background: rgba(88,98,94,.75);
$text-color: rgb(223, 213, 195);

#all-recipe-container {
  display: flex;
  flex-direction: column;
  font-family: 'PT Serif', serif;
  overflow-y: scroll;
  height: 100vh;
}

.recipe-card-container {
  font-family: fantasy, 'sans-serif';
  letter-spacing: 1px;  
  width: 558px;
  margin: 10px auto;
  border-radius: 10px;
  background-color: $recipe-background;
  padding: 10px;
  color: $text-color;
  position: relative;
}

article {
  height: 111px;
  overflow-y: hidden
}

.ingredient img {
  border: 1px solid $recipe-background;
  display: block;
  margin: auto;
}

.ingredient {
  text-align: center;
  width: 100px;
}

.ingredients h3 {
  font-size: 20px;
  margin: 10px 20px;
}

.ingredients {
  padding-top: 33px;
}

.active-recipe {
  height: fit-content;
  
  .ingredients {
    padding-top: 0px;
  }
}

.recipe-details {
  display: flex;
  flex-direction: row;
}

.name-hearts {
  display: flex;
  flex-direction: column;
}

.recipe-name,
.recipe-hearts {
  margin-top: 0px;
  margin-bottom: 5px;
  padding-left: 20px;
}

.recipe-hearts {
  align-items: start;
  display: grid;
  height: 20px;
  grid-template-columns: 35px auto 40px auto;
  width: fit-content;
}

.recipe-hearts p {
  margin: 0;
}

#green-rupee {
  padding-left: 15px;
  width: 30px;
}

.recipe-hearts img {
  width: 25px;
  height: 20px;
}

.recipe-notes {
  margin-left: 10px;
  padding-top: 12px;
}

.recipe-type {
  margin-top: 5px;
  padding-left: 10px;
  margin-left: auto;
}

img {
  width: 80px;
  height: 80px;
}

.ingredients,
.recipe-specific-details {
  display: flex;
  flex-flow: row wrap;
  font-size: 20px;
  justify-content: center;
}

.strength, 
.duration {
  margin-bottom: 0;
  padding: 0 15px;
}

.type_image {
  height: 30px;
  width: 30px;
  background-color: black;
  position: absolute;
  right: 10px;
  top: 10px
}