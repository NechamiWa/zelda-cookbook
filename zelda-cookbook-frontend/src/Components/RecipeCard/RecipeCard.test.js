/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import RecipeCard from './RecipeCard';

describe('RecipeCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<RecipeCard
      id = {106}
      name = 'Baked Apple'
      hearts = 'Â¾'
      type = 'Restore Hearts'
      duration = {null}
      notes = 'Open flame.'
      ingredient1 = 'Apple'
      ingredient2 = {null}
      ingredient3 = {null}
      ingredient4 = {null}
      ingredient5 = {null}
      ingredient1_id = {2}
      ingredient2_id = {null}
      ingredient3_id = {null}
      ingredient4_id = {null}
      ingredient5_id = {null}
      strength = {null}
      resale = '0'
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });
});