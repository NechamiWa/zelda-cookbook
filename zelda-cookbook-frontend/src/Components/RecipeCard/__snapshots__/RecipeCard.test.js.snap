// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCard should match the snapshot 1`] = `
<div
  className="recipe-card-container"
>
  <article
    className="Baked Apple"
    id={106}
    onClick={[Function]}
  >
    <div
      className="Baked Apple recipe-details"
    >
      <img
        alt="Baked Apple"
      />
      <div
        className="Baked Apple name-hearts"
      >
        <h2
          className="Baked Apple recipe-name"
        >
          Baked Apple
        </h2>
        <h3
          className="Baked Apple recipe-hearts"
        >
          <img
            alt="Number Of Hearts"
            src="heart.png"
          />
          <span
            id="targetHeart"
          >
            Â¾
          </span>
          <img
            alt="Green Rupees Jewel"
            id="green-rupee"
            src="GreenRupee.png"
          />
          <p>
            0
          </p>
        </h3>
      </div>
    </div>
    <h3
      className="Baked Apple recipe-notes"
    >
      Notes: 
      Open flame.
    </h3>
    <div
      className="Baked Apple ingredients"
    />
    <div
      className="Baked Apple recipe-specific-details"
    >
      <h5
        className="Baked Apple strength"
      />
      <h5
        className="Baked Apple duration"
      />
    </div>
  </article>
</div>
`;
